### PR TITLE
Add save method for TextDataset

### PR DESCRIPTION
**Pull request by yongtang**
This is an experimental implementation of the `save` (write) method for TextDataset in order to allow saving the content of Dataset into files.

TextDataset itself is straightforward so it may not be that useful. However, it setup an example for saving to other future format (e.g., Parquet/Arrow/etc).

This PR is related to #315

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tests/test_text_eager.py [hunk 1]
```diff
@@ -102,5 +102,30 @@ def on_predict_batch_end(self, batch, logs=None):
...
+  text_io.TextDataset.save(text_dataset, filename)
+
+  with open(filename, 'rb') as f:
+    saved_lines = [line.strip() for line in f]
+  for i, _ in enumerate(saved_lines):
+    assert saved_lines[i] == lines[
+        (2 * i) % len(lines)] + lines[
+            (2 * i + 1) % len(lines)]
+  assert len(saved_lines) == len(lines)
+
```
Comment (BryanCutler) [inline tests/test_text_eager.py]: does the file need to be deleted after the test?

**Comment by yongtang** [comment]
@BryanCutler @terrytangyuan @feihugis Overall the code is ready for review. The usage is similar with pandas (except pandas does not have read_text):
```
import tensorflow_io.frame as frame_io

# Write first 5 (head) entries to dest_path

df = frame_io.read_text(source_path)
df.head().to_text(dest_path)
```

If this is OK I am planning to expand the similar usage to more interesting formats such as csv/parquet.

(The file lock issue has not been resolved though this will be done in follow up PRs).

**Comment by jiachengxu** [comment]
This is a really good example of implementing `save` method, I think I can use the same idea for the JsonDataset

**Comment by yongtang** [comment]
@jiachengxu In PR  #351 I made some enhancement to utilize C++ template. It should be even easier to implement (only need to write a class with `Status Write(WritableFile *file, std::vector<Tensor>& components)` method.

**Pull request merged**